test/cql-pytest/test_materialized_view.py

```diff
@@ -992,3 +992,209 @@ def test_base_clustering_prefix_deletion(cql, test_keyspace):
+            assert [1,2] == sorted([x.p for x in cql.execute(f"SELECT p FROM {mv2} WHERE c=1")])
+            assert [1,2] == sorted([x.p for x in cql.execute(f"SELECT p FROM {mv2} WHERE c=2")])
+
+            cql.execute(f"DELETE FROM {table} WHERE p=1")
+
+            # After the deletion, all data should be gone from both base and view
+            assert [] == list(cql.execute(f"SELECT c FROM {table} WHERE p=1"))
+            assert [] == list(cql.execute(f"SELECT c FROM {mv1} WHERE p=1"))
+            assert [] == list(cql.execute(f"SELECT c FROM {mv2} WHERE p=1 ALLOW FILTERING"))
+            assert [] == list(cql.execute(f"SELECT c FROM {mv3} WHERE p=1 ALLOW FILTERING"))
+
+            assert allN == [x.c for x in cql.execute(f"SELECT c FROM {table} WHERE p=2")]
+            assert allN == sorted([x.c for x in cql.execute(f"SELECT c FROM {mv1} WHERE p=2")])
+
+            for i in range(N):
+                assert [2] == sorted([x.p for x in cql.execute(f"SELECT p FROM {mv2} WHERE c={i}")])
+
+            for i in range(N):
+                assert [v] == sorted([x.v for x in cql.execute(f"SELECT v FROM {mv3} WHERE p=2 AND c={i}")])
+
+# The partition key of the view is strictly contained in the base partition key,
+# so multiple base partitions are combined into one view partition.
+# Test deleting a base partition and verify it is deleted correctly in the view.
+def test_base_partition_deletion_with_smaller_view_partition_key(cql, test_keyspace):
+    with new_test_table(cql, test_keyspace, 'p1 int, p2 int, c int, primary key ((p1,p2),c)') as table:
+        insert = cql.prepare(f'INSERT INTO {table} (p1,p2,c) VALUES (?,?,?)')
+        with new_materialized_view(cql, table, '*', 'p1,p2,c', 'p1 is not null and p2 is not null and c is not null') as mv:
+            # Insert into two separate base partitions.
+            # In the view, the rows have the same partition key.
+            cql.execute(insert, [0, 0, 10])
+            cql.execute(insert, [0, 1, 20])
+
+            # Delete one of the partitions
+            cql.execute(f"DELETE FROM {table} WHERE p1=0 AND p2=0")
+
+            assert [] == list(cql.execute(f"SELECT c FROM {table} WHERE p1=0 AND p2=0"))
+            assert [20] == [x.c for x in cql.execute(f"SELECT c FROM {table} WHERE p1=0 AND p2=1")]
+
+            assert [(1,20)] == [(x.p2, x.c) for x in cql.execute(f"SELECT p2, c FROM {mv} WHERE p1=0")]
+
+# Test deleting a large partition when there is a view with the same partition
+# key, and verify that view updates metrics is increased by exactly 1. Deleting
+# a partition in this case is expected to generate one view update for deleting
+# the corresponding view partition by a partition tombstone.
+def test_base_partition_deletion_with_same_view_partition_key_metrics(cql, test_keyspace, scylla_only):
+    with new_test_table(cql, test_keyspace, 'p int, c int, v int, primary key (p,c)') as table:
+        # Insert into two base partitions. We will delete one of them
```

Comment:
This comment is copy-pasta - this test doesn't actually have base partitions, just one (it's enough).